pkgs/development/libraries/science/math/petsc/default.nix

```diff
@@ -30,23 +54,31 @@ stdenv.mkDerivation rec {
     patchShebangs .
```

Comment:
Can we make this more precise?

Reply:
Can you help me out please -- what do you mean?

The blurb on the `hot shebang stuff` predates my involvement, and I don't know the petsc build system.

Reply:
> Can you help me out please -- what do you mean?

Instead of trying to patch every file it would be better to only specify the file or directory that really needs patching. This would be a lot faster.

Reply:
Looks like: src systems/Apple lib/petsc/bin setup.py config

@jedbrown would that be missing anything now or in the future?

Reply:
Which is for --host and which for --build I wouldn't want to guess, so maybe just using the paths is ok.

Reply:
The build system itself generally uses `$(PYTHON)` and the like (as set in configure) rather than relying on shebangs. There are some installed utilities in `lib/petsc/bin/` that would need shebangs.

Reply:
You can run `/path/to/my/python configure --options` instead of hacking the script.

Reply:
> Which is for --host and which for --build I wouldn't want to guess, so maybe just using the paths is ok.

You don't need to specify this. Just list the paths you want to patch. Having . is very slow for a lot of files and should be avoided.

Reply:
Tried this and it's working. I'm flying blind here, no guarantees on purity etc.

I found that `lib/petsc/bin/petscmpiexec` uses `csh`, which is not in nix currently.
There are also installed files that contain `/usr/bin/env python` on the first line:

 * result/lib/petsc/bin/configureTAS.py
 * result/lib/petsc/bin/extract.py